### PR TITLE
Allows ghost role static spawning

### PR DESCRIPTION
**Pull request by retlaw34**
## About The Pull Request

Tin

Screenshot_7736

## Why It's Good For The Game

This was shockingly easy 

## Changelog

:cl:
add: You can now spawn in as ghost roles as your selected character
/:cl:

### CODE REVIEW

**Comment by unit0016** [comment]
Two notes, one in code review form, another without.
First: Are ghostrole spawners prefs-equipped by *default*? There's a few roles that come to mind that'd be inappropriate to have it enabled on (caves skeleton, for instance) and I want to make sure there's a check for those in place. Can just be a var if it doesn't already exist.

File: local/code/modules/mob_spawn/mob_spawn.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#define GHOSTROLE_RANDOM_APPEARANCE "Random"
+#define GHOSTROLE_STATIC_APPEARANCE "Current"
```
Comment (distributivgesetz) [inline local/code/modules/mob_spawn/mob_spawn.dm:2]: undefine these

File: local/code/modules/mob_spawn/mob_spawn.dm [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/obj/effect/mob_spawn/ghost_role/
```
Comment (distributivgesetz) [inline local/code/modules/mob_spawn/mob_spawn.dm:4]:
```suggestion
/obj/effect/mob_spawn/ghost_role
```

File: local/code/modules/mob_spawn/mob_spawn.dm [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	return spawned_human
+
+/// Original proc in code/modules/mob_spawn/mob_spawn.dm ~line 39.
+/obj/effect/mob_spawn/create(mob/mob_possessor, newname, is_pref_loaded)
+	var/mob/living/spawned_mob = new mob_type(get_turf(src)) //living mobs only
+	name_mob(spawned_mob, newname)
+	special(spawned_mob, mob_possessor)
+	if(!is_pref_loaded)
+		equip(spawned_mob)
+	return spawned_mob
```
Comment (distributivgesetz) [inline local/code/modules/mob_spawn/mob_spawn.dm:51]: seconding ruby's thought, unmodularize this so we get a clear conflict

**Review by distributivgesetz — CHANGES_REQUESTED** [review]

File: local/code/modules/mob_spawn/mob_spawn.dm [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	if(loadout_enabled)
+		spawned_human.equip_outfit_and_loadout(outfit, spawned_human.client.prefs)
+	else
+		equip(spawned_human)
+
+	return spawned_human
+
+/// Original proc in code/modules/mob_spawn/mob_spawn.dm ~line 39.
+/obj/effect/mob_spawn/create(mob/mob_possessor, newname, is_pref_loaded)
+	var/mob/living/spawned_mob = new mob_type(get_turf(src)) //living mobs only
```
[the remark this answers is not in the document]
Comment (lessthnthree) [reply local/code/modules/mob_spawn/mob_spawn.dm:46]: would this be a candidate to do in the style per effigy_proc.dm / local/local.md?
on a return false, could Do Things

**Comment by retlaw34** [comment]
> Two notes, one in code review form, another without. First: Are ghostrole spawners prefs-equipped by _default_? There's a few roles that come to mind that'd be inappropriate to have it enabled on (caves skeleton, for instance) and I want to make sure there's a check for those in place. Can just be a var if it doesn't already exist.

image

Nope, it's disabled by default. The variables were actually already there, just unused.

File: local/code/modules/mob_spawn/mob_spawn.dm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	name_mob(spawned_mob, newname)
+	special(spawned_mob, mob_possessor)
+	if(!is_pref_loaded)
+		equip(spawned_mob)
+	return spawned_mob
```
Comment (distributivgesetz) [inline local/code/modules/mob_spawn/mob_spawn.dm:43]: Remove this proc

**Review by distributivgesetz — CHANGES_REQUESTED** [review]

**Review by distributivgesetz — APPROVED** [review]

**Pull request merged**